src/components/product/index.js

```diff
@@ -100,33 +100,43 @@ const Product = (props) => {
 
   const addToCart = async (item) => {
     try {
-      const user = localStorage.getItem("uid");
-      const cartRef = doc(db, "carts", user);
-      const productRef = doc(cartRef, "products", `${item.id}`);
+      const user = localStorage.getItem('uid');
+      const cartRef = doc(db, 'carts', user);
+      const productRef = doc(cartRef, 'products', `${item.id}`);
       await setDoc(productRef, { ...item, quantity: 1 });
       setIsadded(true);
       context.fetchCartProducts();
     } catch (error) {
-      console.error("Error adding item to cart:", error);
+      console.error('Error adding item to cart:', error);
     }
   };
 
   const addToWishlist = async (item) => {
-    console.log("addToWishlist");
+    console.log('addToWishlist');
     try {
-      const user = localStorage.getItem("uid");
-      const wishlistRef = doc(db, "wishlists", user);
-      const productRef = doc(wishlistRef, "products", `${item.id}`);
+      const user = localStorage.getItem('uid');
+      const wishlistRef = doc(db, 'wishlists', user);
+      const productRef = doc(wishlistRef, 'products', `${item.id}`);
       await setDoc(productRef, { ...item, quantity: 1 });
       setIsadded(true);
       context.fetchWishlistProducts();
     } catch (error) {
-      console.error("Error adding item to wishlist:", error);
+      console.error('Error adding item to wishlist:', error);
     }
   };
 
   if (loading) {
-    return <div>Loading...</div>;
+    return (
+      <div class="skeleton-container">
+        <div class="skeleton-image"></div>
+        <div class="skeleton-details">
+          <div class="skeleton-line skeleton-line-long"></div>
+          <div class="skeleton-line skeleton-line-long"></div>
+          <div class="skeleton-line skeleton-line-short"></div>
+          <div class="skeleton-line skeleton-line-medium"></div>
+        </div>
+      </div>
+    );
```

Comment:
Could you make it a re-usable component ? So we can use it whenever we need to handle the loading state animation ?

Reply:
I've implemented it. Please have a look.